editor/src/core/shared/project-file-types.ts

```diff
@@ -426,18 +426,21 @@ export interface TextFile {
   type: 'TEXT_FILE'
   fileContents: TextFileContents
   lastSavedContents: TextFileContents | null // it is null when the file is saved
+  lastParseSuccess: ParseSuccess | null
```

Comment:
Is the only time that this is null right at the very start? Once it has become not null, it should never be null again?

Reply:
Yeah, once it has successfully been parsed it shouldn't ever be `null` again